### DISCUSSION TITLE
Break Change! 模块类型导出规范

### DISCUSSION THREAD

**Discussion by janryWang** [alibaba/formily#1801] [Ideas]
之前一直在想用Formily.Core这样的全局类型来构建整个框架类型树，但是感觉阻碍太多了：
第一：Typescript官方并不推荐
第二：实际使用的时候，路径太长，也很丑，影响开发效率
第三：像Reactive/Path/Shared这种非常通用的底层库，其实带不带Formily.XXX意义不大，反而限制了它的定位，给人一种它只能用在formily场景上一样
第四：想打包出bundle dts文件给monaco editor消费，不能同时把global类型也打进去
第五：引入方式与普通API引入方式存在gap，一般用户很难想到
第六：对pnpm不友好，后续如果迁移pnpm模式维护的，隐式依赖类型估计会依赖不到

所以，我感觉可以定一个这样的规范，就拿@formily/core来说，index.ts可以这样写
```ts
export * from './shared/externals'
export * from './models'
export * from './effects'
export * from './types'
```

引入的时候
```ts
import { Models,Form } from '@formily/core'

function useForm(form:Form){

}
```

废除掉global.d.ts

不知道大家有没有其他意见

**Comment by liuweiGL** [comment]
我赞同，虽然我有很多要调整的地方。
> 一开始我就说了 namespace 已经被官方 deprecated 了... :penguin:

**Comment by javahuang** [comment]
我也觉得实际使用的时候路径太长了，看着不太舒服。现在改起来麻烦， release 的时候用着爽啊~

**Comment by janryWang** [comment]
我在最新分支已经把类型从index.ts里导出了，只是还没废弃global.d.ts

**Comment by RihuaChen** [comment]
赞同，之前的有点太长了
